src/lit-html.ts

```diff
@@ -307,11 +307,7 @@ export class NodePart extends Part {
             // Since this is the last part we'll use, set it's endNode to the
             // container's endNode. Setting the value of this part will clean
             // up any residual nodes from a previously longer iterable.
-            const range = document.createRange();
-            range.setStartBefore(itemPart.endNode);
-            range.setEndBefore(this.endNode);
-            range.deleteContents();
-            range.detach();
+            this.clear(itemPart.endNode);
```

Comment:
This looks like a change in behavior. Previously the last itemPart's endNode was removed, and it's endNode property set to the container's endNode. Now it looks like, because of the `let node = startNode.nextSibling!;` line below, that the part's endNode will be left around